### ISSUE TITLE
Connecting with SSH timed out

### ISSUE THREAD

**Issue by xkubrt0** [microsoft/vscode-remote-release#2650]
Hi,

last week I could successfully connect to my remote server, but now it is not working. The server is running linux Ubuntu 18. Am I doing something wrong? Changing any Remote SSH extension setting did not help. Opening ssh from Windows cmd line works correctly.

Version: 1.43.2 (user setup)
Commit: 0ba0ca52957102ca3527cf479571617f0de6ed50
Date: 2020-03-24T07:38:38.248Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363

Name: Remote - SSH
Id: ms-vscode-remote.remote-ssh
Version: 0.51.0

Log from VSCode:

```
[09:53:23.305] Log Level: 1
[09:53:23.310] remote-ssh@0.51.0
[09:53:23.310] win32 x64
[09:53:23.311] SSH Resolver called for "ssh-remote+192.168.100.120", attempt 1
[09:53:23.311] SSH Resolver called for host: 192.168.100.120
[09:53:23.311] Setting up SSH remote "192.168.100.120"
[09:53:23.325] Using commit id "0ba0ca52957102ca3527cf479571617f0de6ed50" and quality "stable" for server
[09:53:23.326] Install and start server if needed
[09:53:27.411] Checking ssh with "ssh -V"
[09:53:27.437] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[09:53:27.439] Running script with connection command: ssh -T -D 52572 192.168.100.120 bash
[09:53:27.441] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[09:53:44.441] Resolver error: Connecting with SSH timed out
[09:53:44.443] TELEMETRY: {"eventName":"resolver","properties":{"outcome":"failure","reason":"Timeout","askedPw":"0","askedPassphrase":"0","asked2fa":"0","askedHostKey":"0","gotUnrecognizedPrompt":"0","remoteInConfigFile":"1"},"measures":{"resolveAttempts":1,"retries":1}}
[09:53:44.444] ------


Log from console:
72e476c956a9: running
Acquiring lock on /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0
de6ed50/vscode-remote-lock.jke.0ba0ca52957102ca3527cf479571617f0de6ed50
\ln /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode
-remote-lock.jke.0ba0ca52957102ca3527cf479571617f0de6ed50.target /home/jke/.vsco
de-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.jke.0b
a0ca52957102ca3527cf479571617f0de6ed50
Found existing installation at /home/jke/.vscode-server/bin/0ba0ca52957102ca3527
cf479571617f0de6ed50...
Found running server...

*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
*

Checking server status on port 35399 with wget
72e476c956a9: start
sshAuthSock====
agentPort==35399==
osReleaseId==ubuntu==
arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/1000==
platform==linux==
72e476c956a9: end
```

**Comment by tomajask** [comment]
Hi,

I'm having the exactly same issue right now. It works when I use VSCode from my Mac, but it doesn't work on my windows VSCode. I can share also my details below:

VSCode:
```
Version: 1.43.2 (user setup)
Commit: 0ba0ca52957102ca3527cf479571617f0de6ed50
Date: 2020-03-24T07:38:38.248Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041
```

Remote - SSH:
```
Name: Remote - SSH
Id: ms-vscode-remote.remote-ssh
Description: Open any folder on a remote machine using SSH and take advantage of VS Code's full feature set.
Version: 0.51.0
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh
```

Logs:
```
Enter passphrase for key 'C:\Users\username/.ssh/dev.win.thomasjaskiewicz':
82c0a6fe4c56: running
Acquiring lock on /home/username/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.username.0ba0ca52957102ca3527cf479571617f0de6ed50
\ln /home/username/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.username.0ba0ca52957102ca3527cf479571617f0de6ed50.target /home/username/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.username.0ba0ca52957102ca3527cf479571617f0de6ed50
Installing to /home/username/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50...
82c0a6fe4c56%%1%%
Downloading with wget
Download complete
82c0a6fe4c56%%2%%
SSH_CONNECTION=xxx.xxx.xxx.xxx 56607 xxx.xxx.xxx.xxx 22
LANG=C.UTF-8
XDG_SESSION_ID=124
USER=username
PWD=/home/username
HOME=/home/username
SSH_CLIENT=xxx.xxx.xxx.xxx 56607 22
MAIL=/var/mail/username
SHELL=/bin/bash
SHLVL=2
VSCODE_AGENT_FOLDER=/home/username/.vscode-server
LOGNAME=username
XDG_RUNTIME_DIR=/run/user/1000
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
_=/usr/bin/printenv
OLDPWD=/home/username/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50
Starting server with command... /home/username/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/server.sh --host=127.0.0.1 --enable-remote-auto-shutdown --disable-telemetry --port=0 &> "/home/username/.vscode-server/.0ba0ca52957102ca3527cf479571617f0de6ed50.log" < /dev/null
Waiting for server log...

*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
*

82c0a6fe4c56: start
sshAuthSock====
agentPort==35551==
osReleaseId==ubuntu==
arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/1000==
platform==linux==
82c0a6fe4c56: end
```

**Comment by roblourens** [comment]
@xkubrt0 So a command like this works fine from the windows cmd?

`echo uname | ssh 192.168.100.120 bash`

@tomajask your issue looks different, you can open a new issue

**Comment by xkubrt0** [comment]
@roblourens  No, the command does not work.
`
C:\Users>echo uname | ssh 192.168.100.120 bash
uname: redundant operand "\r"
For more informantion try "uname --help".
`
But I can successfully open ssh 
`
C:\Users>ssh 192.168.100.120 bash
uname
Linux
`

**Comment by passbt** [comment]
My output looks the same as the reported issue where the install Terminal never closes in VSCode. However, I'm able to open another bash terminal and open a directory on the remote server. Thus, it appears VSCode is functional in all regards, but the install/upgrade.

**Comment by roblourens** [comment]
Sorry, I forgot that doesn't work right win->linux because windows insists on using `\r\n`. But that clearly shows that you were able to access the remote. Does setting `"remote.SSH.useLocalServer": true` make any difference in connecting through vscode?

**Comment by xkubrt0** [comment]
> Sorry, I forgot that doesn't work right win->linux because windows insists on using `\r\n`. But that clearly shows that you were able to access the remote. Does setting `"remote.SSH.useLocalServer": true` make any difference in connecting through vscode?

Yes, this fixed the problem! Thank you

**Comment by roblourens** [comment]
That was previously the default. It caused issues and I disabled it by default. Sorry for the trouble. Could you share a log from a working session, in case it gives me a clue as to why this fixed it for you? Because from this info I still have no idea what the issue was...

**Comment by xkubrt0** [comment]
yes of course, if you want to further analyze, why it is not working without local server, just ask, I am happy to support you. Here is the log:

[10:25:16.973] Log Level: 1
[10:25:16.980] remote-ssh@0.51.0
[10:25:16.980] win32 x64
[10:25:16.981] SSH Resolver called for "ssh-remote+192.168.100.120", attempt 1
[10:25:16.981] SSH Resolver called for host: 192.168.100.120
[10:25:16.981] Setting up SSH remote "192.168.100.120"
[10:25:16.984] Acquiring local install lock: C:\Users\honya\AppData\Local\Temp\vscode-remote-ssh-192.168.100.120-install.lock
[10:25:16.998] Looking for existing server data file at c:\Users\honya\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-192.168.100.120-0ba0ca52957102ca3527cf479571617f0de6ed50-0.51.0\data.json
[10:25:16.999] Using commit id "0ba0ca52957102ca3527cf479571617f0de6ed50" and quality "stable" for server
[10:25:17.000] Install and start server if needed
[10:25:17.007] Checking ssh with "ssh -V"
[10:25:17.047] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[10:25:17.047] Preferring non-windows OpenSSH, skipping
[10:25:17.047] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[10:25:17.073] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[10:25:17.073] Preferring non-windows OpenSSH, skipping
[10:25:17.073] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[10:25:17.190] > OpenSSH_7.9p1, OpenSSL 1.1.1a  20 Nov 2018
[10:25:17.192] Using SSH config file "C:\Users\honya\.ssh\config"
[10:25:17.193] askpass server listening on \\.\pipe\vscode-ssh-askpass-32280cb67f11553f4311d2eb6e502251fd66ef74-sock
[10:25:17.193] Spawning local server with {"ipcHandlePath":"\\\\.\\pipe\\vscode-ssh-askpass-1f85113c885bb70053ce4cf4a22bdfb347b63cba-sock","sshCommand":"C:\\Program Files\\Git\\usr\\bin\\ssh.exe","sshArgs":["-v","-T","-D","53793","-F","C:\\Users\\honya\\.ssh\\config","192.168.100.120"],"dataFilePath":"c:\\Users\\honya\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-vscode-remote.remote-ssh\\vscode-ssh-host-192.168.100.120-0ba0ca52957102ca3527cf479571617f0de6ed50-0.51.0\\data.json"}
[10:25:17.194] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"c:\\Users\\honya\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.51.0\\out\\local-server\\askpass.bat","VSCODE_SSH_ASKPASS_NODE":"C:\\Users\\honya\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe","VSCODE_SSH_ASKPASS_MAIN":"c:\\Users\\honya\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.51.0\\out\\askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"\\\\.\\pipe\\vscode-ssh-askpass-32280cb67f11553f4311d2eb6e502251fd66ef74-sock"}
[10:25:17.198] Spawned 7688
[10:25:17.287] > local-server> Spawned ssh: 9124
[10:25:17.309] stderr> OpenSSH_7.9p1, OpenSSL 1.1.1a  20 Nov 2018
[10:25:17.309] stderr> debug1: Reading configuration data C:\\Users\\honya\\.ssh\\config
[10:25:17.310] stderr> debug1: C:\\Users\\honya\\.ssh\\config line 5: Applying options for 192.168.100.120
[10:25:17.318] stderr> debug1: Connecting to 192.168.100.120 [192.168.100.120] port 22.
[10:25:17.319] stderr> debug1: Connection established.
[10:25:17.326] stderr> debug1: identity file /c/Users/honya/.ssh/id_rsa type 0
[10:25:17.327] stderr> debug1: identity file /c/Users/honya/.ssh/id_rsa-cert type -1
[10:25:17.327] stderr> debug1: identity file /c/Users/honya/.ssh/id_dsa type -1
[10:25:17.327] stderr> debug1: identity file /c/Users/honya/.ssh/id_dsa-cert type -1
[10:25:17.327] stderr> debug1: identity file /c/Users/honya/.ssh/id_ecdsa type -1
[10:25:17.327] stderr> debug1: identity file /c/Users/honya/.ssh/id_ecdsa-cert type -1
[10:25:17.328] stderr> debug1: identity file /c/Users/honya/.ssh/id_ed25519 type -1
[10:25:17.328] stderr> debug1: identity file /c/Users/honya/.ssh/id_ed25519-cert type -1
[10:25:17.328] stderr> debug1: identity file /c/Users/honya/.ssh/id_xmss type -1
[10:25:17.328] stderr> debug1: identity file /c/Users/honya/.ssh/id_xmss-cert type -1
[10:25:17.328] stderr> debug1: Local version string SSH-2.0-OpenSSH_7.9
[10:25:17.329] stderr> debug1: Remote protocol version 2.0, remote software version OpenSSH_7.6p1 Ubuntu-4ubuntu0.3
[10:25:17.329] stderr> debug1: match: OpenSSH_7.6p1 Ubuntu-4ubuntu0.3 pat OpenSSH_7.0*,OpenSSH_7.1*,OpenSSH_7.2*,OpenSSH_7.3*,OpenSSH_7.4*,OpenSSH_7.5*,OpenSSH_7.6*,OpenSSH_7.7* compat 0x04000002
[10:25:17.329] stderr> debug1: Authenticating to 192.168.100.120:22 as 'jke'
[10:25:17.332] stderr> debug1: SSH2_MSG_KEXINIT sent
[10:25:17.332] stderr> debug1: SSH2_MSG_KEXINIT received
[10:25:17.332] stderr> debug1: kex: algorithm: curve25519-sha256
[10:25:17.333] stderr> debug1: kex: host key algorithm: ecdsa-sha2-nistp256
[10:25:17.333] stderr> debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
[10:25:17.333] stderr> debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
[10:25:17.335] stderr> debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
[10:25:17.378] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:10wDzPHL9/421VfRsa74Dlp+fWbedwPlwNwunF1bZwA
[10:25:17.379] stderr> debug1: Host '192.168.100.120' is known and matches the ECDSA host key.
[10:25:17.379] stderr> debug1: Found key in /c/Users/honya/.ssh/known_hosts:4
[10:25:17.380] stderr> debug1: rekey after 134217728 blocks
[10:25:17.380] stderr> debug1: SSH2_MSG_NEWKEYS sent
[10:25:17.381] stderr> debug1: expecting SSH2_MSG_NEWKEYS
[10:25:17.381] stderr> debug1: SSH2_MSG_NEWKEYS received
[10:25:17.381] stderr> debug1: rekey after 134217728 blocks
[10:25:17.381] stderr> debug1: Will attempt key: /c/Users/honya/.ssh/id_rsa RSA SHA256:xxxxxxxx
[10:25:17.381] stderr> debug1: Will attempt key: /c/Users/honya/.ssh/id_dsa 
[10:25:17.381] stderr> debug1: Will attempt key: /c/Users/honya/.ssh/id_ecdsa 
[10:25:17.381] stderr> debug1: Will attempt key: /c/Users/honya/.ssh/id_ed25519 
[10:25:17.381] stderr> debug1: Will attempt key: /c/Users/honya/.ssh/id_xmss 
[10:25:17.381] stderr> debug1: SSH2_MSG_EXT_INFO received
[10:25:17.381] stderr> debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
[10:25:17.425] stderr> debug1: SSH2_MSG_SERVICE_ACCEPT received
[10:25:17.426] stderr> debug1: Authentications that can continue: publickey,password
[10:25:17.426] stderr> debug1: Next authentication method: publickey
[10:25:17.427] stderr> debug1: Offering public key: /c/Users/honya/.ssh/id_rsa RSA SHA256:xxxx
[10:25:17.428] stderr> debug1: Server accepts key: /c/Users/honya/.ssh/id_rsa RSA SHA256:xxxx
[10:25:17.431] stderr> debug1: Authentication succeeded (publickey).
[10:25:17.431] stderr> Authenticated to 192.168.100.120 ([192.168.100.120]:22).
[10:25:17.431] stderr> debug1: Local connections to LOCALHOST:53793 forwarded to remote address socks:0
[10:25:17.431] stderr> debug1: Local forwarding listening on ::1 port 53793.
[10:25:17.431] stderr> debug1: channel 0: new [port listener]
[10:25:17.431] stderr> debug1: Local forwarding listening on 127.0.0.1 port 53793.
[10:25:17.431] stderr> debug1: channel 1: new [port listener]
[10:25:17.431] stderr> debug1: channel 2: new [client-session]
[10:25:17.431] stderr> debug1: Requesting no-more-sessions@openssh.com
[10:25:17.432] stderr> debug1: Entering interactive session.
[10:25:17.432] stderr> debug1: pledge: network
[10:25:17.898] stderr> debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
[10:25:17.941] > Welcome to Ubuntu 18.04.4 LTS (GNU/Linux 4.15.0-88-generic x86_64)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> 
>  * Kubernetes 1.18 GA is now available! See https://microk8s.io for docs or
>    install it with:
> 
>      sudo snap install microk8s --channel=1.18 --classic
> 
>  * Multipass 1.1 adds proxy support for developers behind enterprise
>    firewalls. Rapid prototyping for cloud operations just got easier.
> 
>      https://multipass.run/
> 
>  * Canonical Livepatch is available for installation.
>    - Reduce system reboots and improve kernel security. Activate at:
>      https://ubuntu.com/livepatch
> 
> 0 balíků může být aktualizováno.
> 0 aktualizací jsou bezpečnostní aktualizace.
[10:25:17.946] > ready: 563dd422032b
[10:25:17.952] > Linux 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020
[10:25:17.952] Platform: linux
[10:25:17.969] > 563dd422032b: running
[10:25:17.994] > Acquiring lock on /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.jke.0ba0ca52957102ca3527cf479571617f0de6ed50
[10:25:17.996] > \ln /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.jke.0ba0ca52957102ca3527cf479571617f0de6ed50.target /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/vscode-remote-lock.jke.0ba0ca52957102ca3527cf479571617f0de6ed50
[10:25:18.000] > Found existing installation at /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50...
[10:25:18.022] > SSH_CONNECTION=192.168.100.115 53795 192.168.100.120 22
> LANG=cs_CZ.UTF-8
> XDG_SESSION_ID=92
> USER=jke
> PWD=/home/jke
> HOME=/home/jke
> SSH_CLIENT=192.168.100.115 53795 22
> XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
> MAIL=/var/mail/jke
> SHELL=/bin/bash
> SHLVL=2
> VSCODE_AGENT_FOLDER=/home/jke/.vscode-server
> LOGNAME=jke
> DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
> XDG_RUNTIME_DIR=/run/user/1000
> PATH=/home/jke/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
> _=/usr/bin/printenv
> Starting server with command... /home/jke/.vscode-server/bin/0ba0ca52957102ca3527cf479571617f0de6ed50/server.sh --host=127.0.0.1 --enable-remote-auto-shutdown  --port=0 &> "/home/jke/.vscode-server/.0ba0ca52957102ca3527cf479571617f0de6ed50.log" < /dev/null
[10:25:18.023] > Waiting for server log...
[10:25:18.530] >  
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
>  
[10:25:18.536] > 563dd422032b: start
> sshAuthSock====
> agentPort==33141==
> osReleaseId==ubuntu==
> arch==x86_64==
> webUiAccessToken====
> tmpDir==/run/user/1000==
> platform==linux==
> 563dd422032b: end
[10:25:18.537] Received install output: 
sshAuthSock====
agentPort==33141==
osReleaseId==ubuntu==
arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/1000==
platform==linux==

[10:25:18.538] Remote server is listening on port 33141
[10:25:18.538] Parsed server configuration: {"agentPort":33141,"osReleaseId":"ubuntu","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/run/user/1000","platform":"linux"}
[10:25:18.539] Persisting server connection details to c:\Users\honya\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-192.168.100.120-0ba0ca52957102ca3527cf479571617f0de6ed50-0.51.0\data.json
[10:25:18.543] Starting forwarding server. localPort 53799 -> socksPort 53793 -> remotePort 33141
[10:25:18.544] Forwarding server listening on 53799
[10:25:18.544] Waiting for ssh tunnel to be ready
[10:25:18.545] [Forwarding server 53799] Got connection 0
[10:25:18.546] Tunneled remote port 33141 to local port 53799
[10:25:18.547] Resolved "ssh-remote+192.168.100.120" to "127.0.0.1:53799"
[10:25:18.549] TELEMETRY: {"eventName":"resolver","properties":{"outcome":"success","osReleaseId":"ubuntu","arch":"x86_64","askedPw":"0","askedPassphrase":"0","asked2fa":"0","askedHostKey":"0","remoteInConfigFile":"1","gotUnrecognizedPrompt":"0","dynamicForwarding":"1","localServer":"1"},"measures":{"resolveAttempts":1,"retries":1}}
[10:25:18.551] ------




[10:25:18.597] [Forwarding server 53799] Got connection 1
[10:25:18.598] [Forwarding server 53799] Got connection 2
[10:25:19.458] [Forwarding server 53799] Got connection 3
[10:25:29.995] TELEMETRY: {"eventName":"extraInfo","properties":{"remoteCloud":""},"measures":{}}

**Comment by hongyuK** [comment]
Hi，I have the same issue about the remote-ssh. Once I connected with the remote server, the terminal reminded me of inputting the password again and again. I have tried deleting the ~/.vscode-server/bin/2aae1f26c72891c399f860409176fe435a154b13, however, it seems that it didn't work in the way what you described above. The terminal displayed as followed:

[11:49:11.834] Log Level: 2
[11:49:11.848] remote-ssh@0.50.1
[11:49:11.849] win32 x64
[11:49:11.851] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a226a757079746572536572766572227d", attempt 1
[11:49:11.852] SSH Resolver called for host: jupyterServer
[11:49:11.852] Setting up SSH remote "jupyterServer"
[11:49:11.857] Acquiring local install lock: C:\Users\khy\AppData\Local\Temp\vscode-remote-ssh-jupyterServer-install.lock
[11:49:11.900] Looking for existing server data file at c:\Users\khy\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-jupyterServer-2aae1f26c72891c399f860409176fe435a154b13-0.50.1\data.json
[11:49:11.901] Using commit id "2aae1f26c72891c399f860409176fe435a154b13" and quality "stable" for server
[11:49:11.903] Install and start server if needed
[11:49:11.913] Checking ssh with "ssh -V"
[11:49:12.157] > OpenSSH_for_Windows_8.0p1, LibreSSL 2.6.5
[11:49:12.158] Checking ssh with "C:\Windows\System32\OpenSSH\ssh.exe -V"
[11:49:12.161] Got error from ssh: spawn C:\Windows\System32\OpenSSH\ssh.exe ENOENT
[11:49:12.161] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[11:49:12.163] Got error from ssh: spawn C:\Program Files\Git\usr\bin\ssh.exe ENOENT
[11:49:12.163] Checking ssh with "C:\Program Files (x86)\Git\usr\bin\ssh.exe -V"
[11:49:12.164] Got error from ssh: spawn C:\Program Files (x86)\Git\usr\bin\ssh.exe ENOENT
[11:49:12.165] Checking ssh with "ssh -V"
[11:49:12.308] > OpenSSH_for_Windows_8.0p1, LibreSSL 2.6.5
[11:49:12.310] Using SSH config file "C:\Users\khy\.ssh\config"
[11:49:12.310] askpass server listening on \\.\pipe\vscode-ssh-askpass-fd591d386caa1f60f240a242643bb5536b11b564-sock
[11:49:12.311] Spawning local server with {"ipcHandlePath":"\\\\.\\pipe\\vscode-ssh-askpass-8b88d8b2fe0f46ea78a740050dd8ce95786a0ed3-sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","59489","-F","C:\\Users\\khy\\.ssh\\config","jupyterServer"],"dataFilePath":"c:\\Users\\khy\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-vscode-remote.remote-ssh\\vscode-ssh-host-jupyterServer-2aae1f26c72891c399f860409176fe435a154b13-0.50.1\\data.json"}
[11:49:12.311] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"c:\\Users\\khy\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.50.1\\out\\local-server\\askpass.bat","VSCODE_SSH_ASKPASS_NODE":"F:\\新建文件夹\\software\\Microsoft VS Code\\Code.exe","VSCODE_SSH_ASKPASS_MAIN":"c:\\Users\\khy\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.50.1\\out\\askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"\\\\.\\pipe\\vscode-ssh-askpass-fd591d386caa1f60f240a242643bb5536b11b564-sock"}
[11:49:12.326] Spawned 19380
[11:49:12.595] > local-server> Spawned ssh: 2984
[11:49:12.690] stderr> OpenSSH_for_Windows_8.0p1, LibreSSL 2.6.5
[11:49:13.791] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:vyDQn2rvtKZzSHt6ZA4W77rR4d9ocPXEj00Lcy17pi0
[11:49:14.443] Got askpass request: {"request":""}
[11:49:14.444] Listening for interwindow password on \\.\pipe\vscode-ssh-askpass-0d5afc6a22808cf0d4f3f1b8e9c22b01d038697a-sock
[11:49:14.444] Writing password prompt to globalState
[11:49:22.182] Interactor gave response: *********
[11:49:22.183] Cleaning up other-window auth server
[11:49:22.331] stderr> Authenticated to xxxxxxx.
[11:49:22.716] > Welcome to Ubuntu 18.04.2 LTS (GNU/Linux 4.15.0-91-generic x86_64)
> 
>  * Documentation:  https://help.ubuntu.com
>  * Management:     https://landscape.canonical.com
>  * Support:        https://ubuntu.com/advantage
> 
>  System information disabled due to load higher than 20.0
> 
>  * Kubernetes 1.18 GA is now available! See https://microk8s.io for docs or
>    install it with:
> 
>      sudo snap install microk8s --channel=1.18 --classic
> 
>  * Multipass 1.1 adds proxy support for developers behind enterprise
>    firewalls. Rapid prototyping for cloud operations just got easier.
> 
>      https://multipass.run/
> 
>  * Canonical Livepatch is available for installation.
>    - Reduce system reboots and improve kernel security. Activate at:
>      https://ubuntu.com/livepatch
> 
> 173 packages can be updated.
> 0 updates are security updates.
> 
> 
> *** System restart required ***
[11:49:22.809] > ready: 30c44b4df728
[11:49:22.904] > Linux 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020
[11:49:22.905] Platform: linux
[11:49:23.060] > 30c44b4df728: running
[11:49:23.095] > Acquiring lock on /home/khy/.vscode-server/bin/2aae1f26c72891c399f860409176fe435a154b13/vscode-remote-lock.khy.2aae1f26c72891c399f860409176fe435a154b13
[11:49:23.098] > \ln /home/khy/.vscode-server/bin/2aae1f26c72891c399f860409176fe435a154b13/vscode-remote-lock.khy.2aae1f26c72891c399f860409176fe435a154b13.target /home/khy/.vscode-server/bin/2aae1f26c72891c399f860409176fe435a154b13/vscode-remote-lock.khy.2aae1f26c72891c399f860409176fe435a154b13
[11:49:23.105] > Installing to /home/khy/.vscode-server/bin/2aae1f26c72891c399f860409176fe435a154b13...
> 30c44b4df728%%1%%
[11:49:23.110] > Downloading with wget
[11:49:24.643] > Download complete
> 30c44b4df728%%2%%
